SleepEngine/Engine/Components/Transform2D.h

```diff
@@ -8,6 +8,11 @@ class Transform2D : public Component
 {
 public:
     using Base = Component;
+    struct RelativeTransform
+    {
+        glm::vec2 Position{ 0.f, 0.f };
+        float Rotation{ 0.f };
+    };
```

Comment:
Why nest this. It is meant to be used outside of this class, sure they are releated, but not as much. Putting it in the slp namespace would clean code a lot.

Reply:
Moved to a different file. Let it be like this. In the future, I will do caching for global data and we will get rid of this struct.